hangman.py

```diff
@@ -0,0 +1,200 @@
+            else:
+                letters_guessed, guesses_point=letter_in_word(secret_word, letter, list_vowels, letters_guessed, guesses_point, letters_guessed)
+        if is_word_guessed(secret_word, letters_guessed) == 0:
+            final_points = guesses_point*len(set(secret_word))
+            print("------")
+            print("Congratulations, you won!")
+            print("Word:", secret_word)
+            print("Your total score for this game is:", final_points)
+            guesses_point = -1
+        elif is_word_guessed(secret_word, letters_guessed) != 0 and guesses_point <= 0:
+            print("------")
+            print("Sorry, you ran out of guessed. The word was", secret_word)
+            break
+
+
+def match_with_gaps(my_word, other_word):
+    '''
+    my_word: string with _ characters, current guess of secret word
+    other_word: string, regular English word
+    returns: boolean, True if all the actual letters of my_word match the
+        corresponding letters of other_word, or the letter is the special symbol
+        _ , and my_word and other_word are of the same length;
+        False otherwise:
+    '''
+    guesses_word = my_word.replace(" ", "")
+    if len(guesses_word) == len(other_word):
+        for i in range(len(guesses_word)):
+            if guesses_word[i] != '_' and guesses_word[i] != other_word[i]:
+                return False
+        return True
+    else:
+        return False
+
+
+def show_possible_matches(my_word):
+    '''
+    my_word: string with _ characters, current guess of secret word
+    returns: nothing, but should print out every word in wordlist that matches my_word
+             Keep in mind that in hangman when a letter is guessed, all the positions
+             at which that letter occurs in the secret word are revealed.
+             Therefore, the hidden letter(_ ) cannot be one of the letters in the word
+             that has already been revealed.
+    '''
+    list_with_words = []
+    for word in list(wordlist):
+        if match_with_gaps(my_word, word) is True:
+            list_with_words.append(word)
+    print(" ".join(list_with_words))
+
+
+def start():
+    print("Welcome to the game Hangman!\nDo you want play hangman (classic) or hangman with hints?")
+    while type:
+        request = input("if you want to play hangman (classic) input 'classic', play hangman with hints input 'hints':").replace(" ", "")
+        if request == "classic" or request == "hints":
+            if request == 'hints':
+                request = True
+            else:
+                request = False
+            secret_word = choose_word(wordlist)
+            hangman(secret_word, request, list_vowels, warnings_point, letters_guessed, guesses_point)
+            break
+        else:
+            print("Oops! Your input is not correct")
+
+
+if __name__ == "__main__":
+    guesses_point = 6
+    warnings_point = 3
+    letters_guessed = []
```

Comment:
How you pass these variables inside `start` function?

Reply:
I don't understand, what you mean. What do I need to change in this part of code ?